src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -362,5 +364,32 @@ private static void AssertFloatingPointBehavior<T>(T netfxExpectedValue, T netco
                 Assert.Equal(netcoreExpectedValue, testCode());
             }
         }
+
+        // NOTE: LongInputString test is constrained to run on Windows and MacOSX because it causes
+        //       problems on Linux due to the way deferred memory allocation works. On Linux, the allocation can
+        //       succeed even if there is not enough memory but then the test may get killed by the OOM killer at the
+        //       time the memory is accessed which triggers the full memory allocation.
+        private const int MaxExpansionFactorWhileTranscoding = 3;
+        private const int MaxArrayLengthBeforeCalculatingSize = int.MaxValue / 2 / MaxExpansionFactorWhileTranscoding;
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
+        [ConditionalTheory(nameof(IsX64))]
+        [InlineData(MaxArrayLengthBeforeCalculatingSize - 3)]
+        [InlineData(MaxArrayLengthBeforeCalculatingSize - 2)]
+        [InlineData(MaxArrayLengthBeforeCalculatingSize - 1)]
+        [InlineData(MaxArrayLengthBeforeCalculatingSize)]
+        [InlineData(MaxArrayLengthBeforeCalculatingSize + 1)]
+        [InlineData(MaxArrayLengthBeforeCalculatingSize + 2)]
+        [InlineData(MaxArrayLengthBeforeCalculatingSize + 3)]
+        [OuterLoop]
+        public static void LongInputString(int length)
+        {
+            // Verify boundary conditions in Deserialize() that inspect the size to determine allocation strategy.
+            string repeated = new string('x', length - 2);
+            string json = $"\"{repeated}\"";
+            Assert.Equal(length, json.Length);
+
+            string str = JsonSerializer.Deserialize<string>(json);
```

Comment:
nit: Can we add calls to other overloads of Deserialize for this test? Like the one that takes `ROS<byte>` as UTF-8 text input?

Reply:
This is testing a very specific boundary case \ code branch for string only (not byte). 

Are you concerned about large input?

Reply:
> Are you concerned about large input?

Yes, and primarily the async overload that reads from a stream.